static/js/impala/suggestions.js

```diff
@@ -130,12 +130,14 @@ $.fn.searchSuggestions = function($results, processCallback, searchType) {
             $results.filter('.visible').removeClass('visible');
             return;
         }
+        var urlVal = encodeURIComponent($self.val());
```

Comment:
If we really need to escape everything, `encodeURIComponent(str).replace(/'/g, escape)` does it.